### ISSUE TITLE
`ExpansionTile` properties aren't updated with `setState`

### ISSUE THREAD

**Issue by pzierahn** [flutter/flutter#24493]
## Steps to Reproduce

1. Create a new ExpansionTile and set a backgroundColor.
2. Change the ExpansionTile's backgroundColor with setState.

```dart
import 'dart:math';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: MyHomePage());
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  /// Background color for the ExpansionTile
  Color _backgroundColor = randomColor();

  /// Generate random color for the background
  static Color randomColor() => new Color(new Random().nextInt(0xffffffff));

  /// Change event from the ExpansionTile
  void _onExpansion(bool value) {
    /// Change background color. The ExpansionTile doesn't change to the new
    /// _backgroundColor value. The Text element does.
    setState(() {
      _backgroundColor = randomColor();
    });

    print("_onExpansion: value=" + value.toString());
    print("_onExpansion: _backgroundColor=" + _backgroundColor.toString());
  }

  @override
  Widget build(BuildContext context) {
    List<Widget> children = <Widget>[
      /// This view updates to _backgroundColor.
      Text("_backgroundColor=" + _backgroundColor.toString(),
          style: TextStyle(
            color: _backgroundColor,
          )),

      /// The ExpansionTile doesn't change to the new color value.
      ExpansionTile(
        title: Text("_backgroundColor=" + _backgroundColor.toString(),
            style: TextStyle(color: Color(0xff000000))),

        /// backgroundColor: This value will not be updated...
        backgroundColor: _backgroundColor,
        onExpansionChanged: _onExpansion,
        children: <Widget>[
          Text("Text 1"),
          Text("Text 2"),
          Text("Text 3"),
        ],
      )
    ];

    return Scaffold(
        appBar: AppBar(title: Text('Flutter Demo Home Page')),
        body: Center(
            child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: children,
        )));
  }
}
```

## Logs

A log file is attached.
[log.txt](https://github.com/flutter/flutter/files/2592933/log.txt)

### flutter doctor -v
```
[✓] Flutter (Channel beta, v0.11.3, on Mac OS X 10.14.1 18B75, locale en-GB)
    • Flutter version 0.11.3 at /Users/patrick/Tools/flutter
    • Framework revision 72bf075e8d (8 days ago), 2018-11-09 20:36:17 -0800
    • Engine revision 5646e86a6f
    • Dart version 2.1.0 (build 2.1.0-dev.9.3 9c07fb64c4)

[!] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/patrick/Library/Android/sdk/
    • Android NDK at /Users/patrick/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/patrick/Library/Android/sdk/
    • Java binary at: /Applications/Android Studio 3.3 Preview.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    ✗ Android license status unknown.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3
```

**Comment by zoechi** [comment]
I can reproduce.
It seems `backgroundColor` is only updated in `didChangeDependencies()`.

Wrapping with a theme that is updated worked as a workaround for me

```
      Theme(
        data: Theme.of(context).copyWith(backgroundColor: _backgroundColor),
        child: ExpansionTile(
          key: Key('backgroundcolor'),
          title: Text("_backgroundColor=" + _backgroundColor.toString(),
              style: TextStyle(color: Color(0xff000000))),

          /// backgroundColor: This value will not be updated...
          backgroundColor: _backgroundColor,
          onExpansionChanged: _onExpansion,
          children: <Widget>[
            Text("Text 1"),
            Text("Text 2"),
            Text("Text 3"),
          ],
        ),
      )
```

**Comment by liquidiert** [comment]
Did something happen in the meantime? Wrapping with Theme didn't work for me...

**Comment by nicowernli** [comment]
Same here, wrapping with Theme is not doing it.

<img width="934" alt="Captura de pantalla 2019-12-17 a las 11 39 59" src="https://user-images.githubusercontent.com/2279893/70988486-38fbed00-20c2-11ea-84fe-398cb6e728f8.png">

**Comment by nicowernli** [comment]
The "problem" is that the `_backgroundColorTween` starts `transparent` and that's why the Tile doesn't get the background defined in the `backgroundColor` property. The `backgroundColor` is only used as the `end` in the `_backgroundColorTween`.

**Comment by liquidiert** [comment]
nicowernli what do you mean with _backgroundColorTween ?

**Comment by nicowernli** [comment]
@liquidiert those are private properties of the `ExpansionTile`. Check the source code and you'll see them, there is no way to change those values right now.

What I suggest is just wrapping the ExpansionTile with a Container with the desired backgroundColor and also setting the `_backgroundColor` of the `ExpationTile` (this will be applied when the `ExpantionTile` is "open".

```dart
Container(
   color: _backgroundColor,
   child: ExpantionTile(
      backgroundColor: _backgroundColor,
      ...
   ),
)
```

**Comment by liquidiert** [comment]
Ah ok but the original issue and mine is that the color of background when expanded doesn't change up on setState :/

**Comment by TahaTesser** [comment]
<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.20.0-0.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-0.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision d9653445f4 (8 days ago), 2020-06-09 18:43:03 -0400
    • Engine revision e8c13aa012
    • Dart version 2.9.0 (build 2.9.0-14.0.dev 5c1376615e)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (4 available)
    • SM M305F   • 32003c30dc19668f • android-arm64  • Android 10 (API 29)
    • macOS      • macOS            • darwin-x64     • Mac OS X 10.15.5 19F101
    • Web Server • web-server       • web-javascript • Flutter Tools
    • Chrome     • chrome           • web-javascript • Google Chrome
      83.0.4103.97

• No issues found!
``` 

</details>

**Comment by maheshmnj** [comment]
Reproduces as stable 3.0.1 and master 3.1.0 the backgroundColor of `ExpansionTile` does not change in runtime.

<details>
<summary>code sample</summary>

```dart
import 'dart:math';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: MyHomePage());
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Color _backgroundColor = randomColor();

  static Color randomColor() => Color(Random().nextInt(0xffffffff));

  void _onExpansion(bool value) {
    setState(() {
      _backgroundColor =
          Colors.accents[Random().nextInt(Colors.accents.length - 1)];
    });

    print("_onExpansion: value=" + value.toString());
    print("_onExpansion: _backgroundColor=" + _backgroundColor.toString());
  }

  @override
  Widget build(BuildContext context) {
    List<Widget> children = <Widget>[
      Text("_backgroundColor=" + _backgroundColor.toString(),
          style: TextStyle(
            color: _backgroundColor,
          )),
      ExpansionTile(
        title: Text("_backgroundColor=" + _backgroundColor.toString(),
            style: const TextStyle(color: Color(0xff000000))),
        backgroundColor: _backgroundColor,
        onExpansionChanged: _onExpansion,
        children: const <Widget>[
          Text("Text 1"),
          Text("Text 2"),
          Text("Text 3"),
        ],
      )
    ];

    return Scaffold(
        appBar: AppBar(title: const Text('Flutter Demo Home Page')),
        body: Center(
            child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: children,
        )));
  }
}
```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.0.1 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (3 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin
        usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555                     • android-arm64  • Android 11 (API 30)
    • iPhone 12 Pro (mobile) • 19FD0231-BFF0-441D-B584-AD94C4084525 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1138, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.1.0-0.0.pre.1138 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d9be807192 (9 hours ago), 2022-06-06 21:16:49 -0700
    • Engine revision 48864fc837
    • Dart version 2.18.0 (build 2.18.0-170.0.dev)
    • DevTools version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile)    • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • sdk gphone arm64 (mobile) • emulator-5554    • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos            • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)              • chrome           • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by dam-ease** [comment]
This issue is still reproducible on the latest master and stable. The text color changes but the background color of the ExpansionTile doesn't.

<details><summary><code>stable, master flutter doctor -v</code></summary>

```
[!] Flutter (Channel stable, 3.7.12, on macOS 13.3 22E252 darwin-arm64, locale en-NG)
    • Flutter version 3.7.12 on channel stable at /Users/dammya/fvm/versions/3.7.12
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/dammya/fvm/versions/3.7.12. Consider adding /Users/dammya/fvm/versions/3.7.12/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (2 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/dammya/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (3 available)
    • TECNO LC6a (mobile) • 044882598G014637 • android-arm    • Android 9 (API 28)
    • macOS (desktop)     • macos            • darwin-arm64   • macOS 13.3 22E252 darwin-arm64
    • Chrome (web)        • chrome           • web-javascript • Google Chrome 112.0.5615.137

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.10.0-19.0.pre.31, on macOS 13.3 22E252 darwin-arm64, locale en-NG)
    • Flutter version 3.10.0-19.0.pre.31 on channel master at /Users/dammya/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/dammya/fvm/versions/master. Consider adding /Users/dammya/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 110fe75b5d (4 hours ago), 2023-05-03 02:11:35 -0400
    • Engine revision 0407a16a9b
    • Dart version 3.1.0 (build 3.1.0-66.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/dammya/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (3 available)
    • TECNO LC6a (mobile) • 044882598G014637 • android-arm    • Android 9 (API 28)
    • macOS (desktop)     • macos            • darwin-arm64   • macOS 13.3 22E252 darwin-arm64
    • Chrome (web)        • chrome           • web-javascript • Google Chrome 112.0.5615.137

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category
```

</details>
